src/platform/ESP32/ESP32EndpointQueueFilter.h

```diff
@@ -0,0 +1,114 @@
+/*
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <inet/EndpointQueueFilter.h>
+#include <lwip/ip4_addr.h>
+
+namespace chip {
+namespace Inet {
+
+class ESP32EndpointQueueFilter : public EndpointQueueFilter
+{
+public:
+    CHIP_ERROR SetMdnsHostName(const chip::CharSpan & hostName)
+    {
+        ReturnErrorCodeIf(hostName.size() != sizeof(mHostNameBuffer), CHIP_ERROR_INVALID_ARGUMENT);
+        ReturnErrorCodeIf(!IsValidMdnsHostName(hostName), CHIP_ERROR_INVALID_ARGUMENT);
+        memcpy(mHostNameBuffer, hostName.data(), hostName.size());
+        return CHIP_NO_ERROR;
+    }
+
+    FilterOutcome FilterBeforeEnqueue(const void * endpoint, const IPPacketInfo & pktInfo,
+                                      const chip::System::PacketBufferHandle & pktPayload) override
+    {
+        if (!IsMdnsBroadcastPacket(pktInfo))
+        {
+            return FilterOutcome::kAllowPacket;
+        }
+        // Drop the mDNS packets which don't contain 'matter' or '<device-hostname>'.
+        const uint8_t matterBytes[] = { 'm', 'a', 't', 't', 'e', 'r' };
+        if (PayloadContains(pktPayload, ByteSpan(matterBytes)) || PayloadContains(pktPayload, ByteSpan(mHostNameBuffer)))
```

Comment:
Note that hostnames are *not* case sensitive, so it's possible a querier will query the lowercase version of the hostname... This should be fine, but is not mDNS compliant. It would be better to check the hostname in both lowercase and uppercase by having a `PayloadContainsCaseInsensitive` matcher that makes A-F/a-f of the hostname buffer equivalent.

Reply:
Added